packages/slate/src/models/point.js

```diff
@@ -120,6 +120,79 @@ class Point extends Record(DEFAULTS) {
     return !this.isSet
   }
 
+  /**
+   * Check whether the point is after a `range`.
+   *
+   * @return {Boolean}
+   */
+
+  isAfterRange(range) {
+    if (this.isUnset) return false
+    const is =
+      (this.key === range.end.key && this.offset > range.end.offset) ||
+      PathUtils.compare(this.path, range.end.path) === 1
+    return is
+  }
+
+  /**
+   * Check whether the point is at the end of a `range`.
+   *
+   * @return {Boolean}
+   */
+
+  isAtEndOfRange(range) {
+    if (this.isUnset) return false
+    const is = this.key === range.end.key && this.offset === range.end.offset
+    return is
+  }
+
+  /**
+   * Check whether the point is at the start of a `range`.
+   *
+   * @return {Boolean}
+   */
+
+  isAtStartOfRange(range) {
+    if (this.isUnset) return false
+    const is =
+      this.key === range.start.key && this.offset === range.start.offset
+    return is
+  }
+
+  /**
+   * Check whether the point is before a `range`.
+   *
+   * @return {Boolean}
+   */
+
+  isBeforeRange(range) {
```

Comment:
Could to do the same for `isBeforePoint(point)` too?

Reply:
:+1: